src/document/doc-debug.js

```diff
@@ -64,7 +68,12 @@ function printDoc(doc) {
   }
 
   if (doc.type === "break-parent") {
-    return "breakParent";
+    const afterHardline =
+      Array.isArray(parentParts) &&
+      parentParts[index - 1] &&
+      parentParts[index - 1].type === "line" &&
+      parentParts[index - 1].hard;
+    return afterHardline ? undefined : "breakParent";
```

Comment:
Do you think it's easier to move this part to `flattenDoc`, add a function to replace `hardline + breakParent` as the new `parts`

https://github.com/thorn0/prettier/blob/2fad7569a9a9205ac222878bf21d5a53ff32cb50/src/document/doc-debug.js#L19
And
https://github.com/thorn0/prettier/blob/2fad7569a9a9205ac222878bf21d5a53ff32cb50/src/document/doc-debug.js#L37